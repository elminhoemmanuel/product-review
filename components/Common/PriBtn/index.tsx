import React from 'react'

type PriBtnProps = React.ComponentProps<'button'>

const PriBtn = ({ children, ...rest }: PriBtnProps) => {
    return (
        <button {...rest} className="text-sm block w-full px-5 py-3 text-ckdarkblue font-bold  bg-ckbtn hover:bg-cklightblue rounded-md">
            {children}
        </button>
    )
}

export default PriBtn