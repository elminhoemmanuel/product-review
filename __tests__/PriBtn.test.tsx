import PriBtn from '@/components/Common/PriBtn';
import { render, screen } from '@testing-library/react'

test("On render, check that Primary button is rendered correctly", () => {
    const { container } = render(
        <PriBtn>Test Button</PriBtn>
    );
    // eslint-disable-next-line testing-library/no-node-access
    expect(container.firstChild).toMatchSnapshot()
})
