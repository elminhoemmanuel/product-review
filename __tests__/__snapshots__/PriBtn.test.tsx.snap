// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`On render, check that Input field is present 1`] = `
<button
  class="text-sm block w-full px-5 py-3 text-ckdarkblue font-bold bg-ckbtn hover:bg-cklightblue rounded-md"
>
  Test Button
</button>
`;

exports[`On render, check that Primary button is rendered correctly 1`] = `
<button
  class="text-sm block w-full px-5 py-3 text-ckdarkblue font-bold bg-ckbtn hover:bg-cklightblue rounded-md"
>
  Test Button
</button>
`;
